Library/Homebrew/cask/cask.rb

```diff
@@ -292,14 +292,20 @@ def artifacts_list
       artifacts.map do |artifact|
         next artifact.to_h if artifact.is_a? Artifact::AbstractFlightBlock
 
+        if artifact.is_a? Artifact::Relocated
+          source, *args = artifact.to_args
+          next { artifact.class.dsl_key => [to_h_string_gsubs(source, replace_prefix: false), *to_h_gsubs(args)] }
+        end
+
         { artifact.class.dsl_key => to_h_gsubs(artifact.to_args) }
```

Comment:
```suggestion
        case artifact
        when Artifact::AbstractFlightBlock
          artifact.to_h
        when Artifact::Relocated
          source, *args = artifact.to_args
          { artifact.class.dsl_key => [to_h_string_gsubs(source, replace_prefix: false), *to_h_gsubs(args)] }
        else
          { artifact.class.dsl_key => to_h_gsubs(artifact.to_args) }
        end
```

Also, it might be good to explain this edge case with a comment.